File: src/algorithms/find-max-k.js
/**
 * Given an integer array nums that does not contain any zeroes, finds the
 * largest positive integer k such that -k also exists in the array,
 * and returns the positive integer k. If no such pairing exists, returns -1.
 * 
 * @param {number[]} nums - An array of positive and negative numbers
 * @returns {number} - Returns a number
 */
function findMaxK (nums) {
  let numsSet = new Set(nums);
  let largestInt = 0;
  for (const num of numsSet) {
    if (num > largestInt && numsSet.has(-num)) {
      largestInt = num;
    }
  }
  if (largestInt === 0) {
    largestInt = -1;
  }
  return largestInt;
}

File: src/algorithms/find-max-k.py
def findMaxK(nums):
    """Given an integer array nums that does not contain any zeros, find the 
    largest positive integer k such that -k also exists in the array. Return 
    the positive integer k. If there is no such integer, return -1.

    Args:
        nums (list[int]): A list of integers.

    Returns:
        int: Returns an integer.
    """
    res = -1
    seen = set()
    for num in nums:
        temp = abs(num)
        if temp > res and -num in seen:
            res = temp
        seen.add(num)
    return res
